lib/whitehall_importer/integrity_checker.rb

```diff
@@ -87,10 +88,72 @@ def organisation_problems
       problems
     end
 
+    def unpublishing_problems
+      problems = []
+      return problems unless edition.withdrawn? || edition.removed?
+
+      unless expected_state?
+        problems << problem_description(
+          "edition state isn't expected",
+          "unpublished",
+          publishing_api_content.dig("publication_state"),
+        )
+        return problems
+      end
+
+      check = UnpublishingCheck.new(edition, publishing_api_unpublishing)
+
+      unless check.expected_type?
+        problems << unpublishing_type_problem(
+          edition.state,
+          publishing_api_unpublishing["type"],
+        )
```

Comment:
I wonder if this could be a problem description e.g: 

```
        problems << problem_description(
          "unpublshing type doesn't match", 
          publishing_api_unpublishing["type"],
          check.expected_unpublishing_type,
        )
```
?

Reply:
Changed this to use `problem_description`.